### ISSUE TITLE
Gen 2, preSignUp lambda trigger unexpected behavior

### ISSUE THREAD

**Issue by johnpc** [aws-amplify/amplify-ui#5000]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

Next.js

### What browsers are you seeing the problem on?

Chrome

### Which region are you seeing the problem in?

us-west-2

### Please describe your bug.

Hi folks, I experienced an issue with preSignUp triggers and I'm not sure where to direct this problem:

1. If using autoConfirmUser in preSignUp, you get "stuck" in the authenticator component after sign up
2. ~The preSignUp lambda executes twice (apparently to due to OPTIONS and POST calls) which is dangerous as setup-y stuff is only supposed to happen once~
3. The preSignUp lambda will always time out unless context.callbackWaitsForEmptyEventLoop = false; is set in the lambda

### What's the expected behaviour?

I would expect to be immediately logged in after signup. I would expect the preSignUp trigger lambda to have been called only once. I would expect the lambda to not timeout if `context.callbackWaitsForEmptyEventLoop = false;` is omitted from the lambda code.

### Help us reproduce the bug!

I have a full working minimal reproduction of these issues here: https://github.com/johnpc/autoconfirm-signup-authenticator-bug which is deployed and interact-able here: https://main.d1q2mvk8vhrxl0.amplifyapp.com

### Code Snippet

```typescript
// Put your code below this line.

```

**Comment by johnpc** [comment]
Screen recording of issue here: 

![Screen Recording](https://github.com/aws-amplify/amplify-ui/assets/4824042/0cd9324c-e174-4de1-9b90-d31f6d1d89ed)

**Comment by calebpollman** [comment]
Was able to repro the issue using the provided example app. Currently digging in to whether the issue is on the JS lib side od the `Authenticator` itself

**Comment by calebpollman** [comment]
Verified the issue is in the `Authenticator`. Working on a fix

**Comment by calebpollman** [comment]
While we are working on a fix, a workaround for the issue can be applied by downgrading the versions of `@aws-amplify/ui-react` and to `aws-amplify` to version `5` in the _package.json_ `dependencies` field:

```json
"aws-amplify": "^5",
"@aws-amplify/ui-react": "^5",
```

**Comment by calebpollman** [comment]
Fix has been released in `@aws-amplify/ui-react@6.1.4`
